algebras/algebra/src/main/scala/endpoints4s/algebra/ChunkedEntities.scala

```diff
@@ -77,24 +92,53 @@ trait ChunkedEntities extends EndpointsWithCustomErrors {
   *
   * @group algebras
   */
-trait ChunkedJsonEntities extends ChunkedEntities with JsonCodecs {
+trait ChunkedJsonEntities extends ChunkedJsonRequestEntities with ChunkedJsonResponseEntities
+
+trait ChunkedJsonRequestEntities extends ChunkedRequestEntities with JsonCodecs with Framing {
+
+  @deprecated(
+    "Use jsonChunksRequest[A](framing: RequestFraming) instead to explicitly provide chunk framing",
+    "1.6.1"
+  )
+  def jsonChunksRequest[A](implicit
+      codec: JsonCodec[A]
+  ): RequestEntity[Chunks[A]]
 
   /** A request entity carrying chunks of JSON values
     *
     * @tparam A Type of values serialized into JSON
     * @group operations
     */
-  def jsonChunksRequest[A](implicit
+  def jsonChunksRequest[A](framing: Framing)(implicit
       codec: JsonCodec[A]
   ): RequestEntity[Chunks[A]]
+}
+
+trait ChunkedJsonResponseEntities extends ChunkedResponseEntities with JsonCodecs with Framing {
+
+  @deprecated(
+    "Use jsonChunksResponse[A](framing: ResponseFraming) instead to explicitly provide chunk framing",
+    "1.6.1"
+  )
+  def jsonChunksResponse[A](implicit
+      codec: JsonCodec[A]
+  ): ResponseEntity[Chunks[A]]
 
   /** A response entity carrying chunks of JSON values
     *
     * @tparam A Type of values serialized into JSON
     * @group operations
     */
-  def jsonChunksResponse[A](implicit
+  def jsonChunksResponse[A](framing: Framing)(implicit
       codec: JsonCodec[A]
   ): ResponseEntity[Chunks[A]]
+}
+
+/** Algebra interface for describing how chunks of chunked transfer-encoding requests and responses should be framed.
+  * Being explicit about how chunks are framed solves the issue of re-framing happening during transport.
+  */
+trait Framing {
 
+  type Framing
```

Comment:
This is just a stub of documentation, it would be good to emphasize that client and server interpreters have to be consistent (so that streams framed by a server are correctly decoded by a client that uses the same `Framing`).